llms/mlx_lm/evaluate.py

```diff
@@ -209,13 +183,45 @@ def loglikelihood(self, requests) -> list[tuple[float, bool]]:
                 + "completion longer than context."
             )
 
+        num_results = len(shortened)
+
+        # sort by length to get batches with little padding.
+        sorted_indices = sorted(range(len(shortened)), key=lambda i: -len(shortened[i]))
+        shortened = [shortened[i] for i in sorted_indices]
+        completion_spans = [completion_spans[i] for i in sorted_indices]
+
+        group = mx.distributed.init() if mx.distributed.is_available() else None
+        if group is not None:
+            # split strided so we have approximately the same lengths on each node
+            shortened = shortened[group.rank() :: group.size()]
+            completion_spans = completion_spans[group.rank() :: group.size()]
```

Comment:
You don't necessarily need to check `is_available()`. If it's not available then `rank = 0` and `size = 1` and everything should work fine. So you can avoid needing to condition code on `group is not None`.